File: client/src/types/order.ts
export type OrderStatus = "pending_payment" | "paid" | "expired" | "cancelled";

export type OrderProductSummary = {
  id: string;
  name: string;
  brand: string | null;
  image_url: string | null;
};

export type Order = {
  id: string;
  listing_id: string;
  buyer_id: string;
  amount: number;
  status: OrderStatus;
  payment_deadline: string;
  razorpay_order_id: string | null;
  razorpay_payment_id: string | null;
  created_at: string;
  paid_at: string | null;
  product: OrderProductSummary | null;
};

export type OrdersPage = {
  items: Order[];
  total: number;
  page: number;
  page_size: number;
};

export type RazorpayOrder = {
  razorpay_order_id: string;
  amount: number;
  currency: string;
  key_id: string;
};

File: server/schemas/order.py
from pydantic import BaseModel


class OrderProductSummary(BaseModel):
    id: str
    name: str
    brand: str | None = None
    image_url: str | None = None


class OrderOut(BaseModel):
    id: str
    listing_id: str
    buyer_id: str
    amount: float
    status: str
    payment_deadline: str
    razorpay_order_id: str | None = None
    razorpay_payment_id: str | None = None
    created_at: str
    paid_at: str | None = None
    product: OrderProductSummary | None = None


class OrdersPage(BaseModel):
    items: list[OrderOut]
    total: int
    page: int
    page_size: int


class RazorpayOrderOut(BaseModel):
    razorpay_order_id: str
    amount: float
    currency: str
    key_id: str


class VerifyPaymentRequest(BaseModel):
    razorpay_order_id: str
    razorpay_payment_id: str
    razorpay_signature: str
